packages/bpk-scrim-utils/src/withScrim.tsx

```diff
@@ -138,9 +144,9 @@ const withScrim = (WrappedComponent) => {
 
       return (
         <div className={classNames.join(' ')}>
-          <BpkScrim onClose={closeOnScrimClick ? onClose : null} />
+          <BpkScrim onClose={closeOnScrimClick ? onClose : undefined} />
           <WrappedComponent
-            {...rest}
+            {...(rest as P)}
```

Comment:
Should we be defining `rest` within our `Props` object or is this due to it being a class component that it is done this way?

Reply:
this is something specific to hocs actually, and I believe the reason you need to cast it like this is actually due to a bug in TS - https://github.com/Microsoft/TypeScript/issues/28938
But I did notice I believe a mistake here so let me fix that quickly 😬